tests/Container/EventStoreFactoryTest.php

```diff
@@ -0,0 +1,151 @@
+use PHPUnit_Framework_TestCase as TestCase;
+use Prooph\Common\Event\ActionEventEmitter;
+use Prooph\EventStore\Adapter\Adapter;
+use Prooph\EventStore\Container\EventStoreFactory;
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\Feature\Feature;
+
+/**
+ * Class EventStoreFactoryTest
+ * @package Prooph\EventStoreTest\Container
+ */
+class EventStoreFactoryTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function it_creates_event_store()
+    {
+        $config['prooph']['event_store'] = [];
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->once())->method('get')->with('config')->willReturn($config);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
+    }
+
+    /**
+     * @test
+     */
+    public function it_injects_custom_adapter()
+    {
+        $config['prooph']['event_store']['adapter'] = 'adapter';
+
+        $adapterMock = $this->getMockForAbstractClass(Adapter::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('adapter')->willReturn($adapterMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
+    }
+
+    /**
+     * @test
+     */
+    public function it_injects_custom_event_emitter()
+    {
+        $config['prooph']['event_store']['event_emitter'] = 'event_emitter';
+
+        $eventEmitterMock = $this->getMockForAbstractClass(ActionEventEmitter::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('event_emitter')->willReturn($eventEmitterMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
+    }
+
+    /**
+     * @test
+     */
+    public function it_injects_plugins()
+    {
+        $config['prooph']['event_store']['plugins'][] = 'plugin';
+
+        $featureMock = $this->getMockForAbstractClass(Feature::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('plugin')->willReturn($featureMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
```

Comment:
same as above

here we can expect that the `setUp` method of the `$featureMock` is inoked